src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -0,0 +1,29 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.ProductDao;
+
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ProductDetailsPageServlet extends HttpServlet {
+    private ProductDao products;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        products = ArrayListProductDao.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String productId = request.getPathInfo();
+
+        request.setAttribute("product",products.getProduct(Long.parseLong(productId.substring(1))));
```

Comment:
Do not believe user input: what if the user passes a string instead of a number in the request?

Reply:
The lecturer suggested to implement general error.jsp page to 500 error-code, so if the user passes in a string, the program will show error.jsp page.